libarchive/archive_read_support_format_7zip.c

```diff
@@ -986,15 +987,27 @@ ppmd_read(void *p)
 	struct _7zip *zip = (struct _7zip *)(a->format->data);
 	Byte b;
 
-	if (zip->ppstream.avail_in == 0) {
-		archive_set_error(&a->archive, ARCHIVE_ERRNO_FILE_FORMAT,
-		    "Truncated RAR file data");
-		zip->ppstream.overconsumed = 1;
-		return (0);
+	if (zip->ppstream.avail_in <= 0) {
+		// Ppmd7_DecodeSymbol might require reading multiple bytes and we are on boundary; last resort to read using __archive_read_ahead.
```

Comment:
Please use multi-line comment, we try to fit to 80 chars:
```c
/*
 * This is my comment line 1
 * This is my comment line 2
 */

Reply:
thank you for showing me the format